### ISSUE TITLE
Allow axis withouth <xyz> in ball joints

### ISSUE THREAD

**Issue by marcoag** [gazebosim/sdformat#1250]
Ball joints don't really have an axis however it is still used in the sdf file to specify dynamics parameters like damping or friction. All axis currently need to specify some `<xyz>` values but since this is of no use for a ball joint it can be confusing for users (i.e: https://github.com/RobotLocomotion/drake/pull/18966). The idea is to remove the requirement for `<xyz>` values for ball joints.

Maybe some clarification to the specification regarding ball joints could be considered too.

**Comment by scpeters** [comment]
Yes, we should clarify the specification for ball joints. If an `<axis>` element does not have `<xyz>`, it will be added with default value `0 0 1`. If we ever implement support for ball joint cone limits (see https://github.com/gazebosim/sdformat/issues/860), then I think the `<xyz>` would be used for specifying the cone direction.
